client/consul/consul.go

```diff
@@ -29,3 +36,101 @@ type ServiceIdentityAPI interface {
 type SupportedProxiesAPI interface {
 	Proxies() (map[string][]string, error)
 }
+
+// JWTLoginRequest is an object representing a login request with JWT
+type JWTLoginRequest struct {
+	JWT            string
+	Role           string
+	AuthMethodName string
+}
+
+// ConsulClient is the interface that the nomad client uses to interact with
+// Consul.
+type ConsulClient interface {
+	// DeriveSITokenWithJWT contacts the nomad server and fetches wrapped tokens
+	// for a set of tasks. The wrapped tokens will be unwrapped using consul and
+	// returned.
+	DeriveSITokenWithJWT(map[string]JWTLoginRequest) (map[string]string, error)
+}
+
+type consulClient struct {
+	// tokenDeriver is a function pointer passed in by the client to derive
+	// tokens by making RPC calls to the nomad server.
+	tokenDeriver TokenDeriverFunc
+
+	// client is the API client to interact with vault
```

Comment:
```suggestion
	// client is the API client to interact with consul
```

Reply:
Changed in [4c925f3](https://github.com/hashicorp/nomad/pull/18370/commits/4c925f3a61b8b612482abe61fc7013dab06fd041)